crates/c-api/tests/store.cc

```diff
@@ -12,6 +12,12 @@ TEST(Store, Smoke) {
 
   store = Store(engine);
   store.limiter(-1, -1, -1, -1, -1);
+  store.epoch_deadline_callback(
+      [](wasmtime::Store::Context /* context */, uint64_t &epoch_deadline_delta)
+          -> wasmtime::Result<wasmtime::DeadlineKind> {
+        epoch_deadline_delta += 1;
+        return wasmtime::DeadlineKind::Continue;
+      });
```

Comment:
To help catch the above bug, would you be up for expanding the test her to run the callback as well?

Reply:
I believe it mostly mirrors the Rust-side API here. Other than that though no particular reasons, you're right that the Rust side could use the store to update the dealine and then return continue/yield as an enum.